### PR TITLE
changes for v3name_fuzzer.

### PR DESCRIPTION
**Pull request by hardik05**
##### Checklist

- [x ] tests are added or updated

### CODE REVIEW

**Comment by hardik05** [comment]
Let me know if it require CLA? i can send and then you can merge.

File: fuzz/v3name_fuzzer.c [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+{
+    return 1;
+}
+
+int FuzzerTestOneInput(const uint8_t* data, size_t size){     
+    GENERAL_NAME *namesa = OPENSSL_malloc(size);
+    GENERAL_NAME *namesb = OPENSSL_malloc(size);
+
+    if (namesa == NULL || namesb == NULL)
+        return 0;
```
Comment (kroeckx) [inline fuzz/v3name_fuzzer.c:27]: There is no need to allocate or check this. The d2i_GENERAL_NAME() below will allocate it, and so the ones above get leaked.

File: fuzz/v3name_fuzzer.c [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+         * we compare them they are always different pointers.
+         */
+        namesa = d2i_GENERAL_NAME(NULL, &derp, size);
+        derp = data;
+        namesb = d2i_GENERAL_NAME(NULL, &derp, size);        
+        GENERAL_NAME_cmp(namesa, namesb);
+        if (namesa != NULL)
+            GENERAL_NAME_free(namesa);
+        if (namesb != NULL)
+            GENERAL_NAME_free(namesb);
```
Comment (kroeckx) [inline fuzz/v3name_fuzzer.c:42]: Most of those lines have too many spaces.

**Comment by kroeckx** [comment]
I suggest you just call it "v3name", drop the fuzzer from the name of the binary.

Note that if you make changes, you can just push your new commits to your branch, there is no need to make a new pull request.

File: fuzz/v3name_fuzzer.c (hunk 1 above)
Comment (hardik05) [reply fuzz/v3name_fuzzer.c]: done.

File: fuzz/v3name_fuzzer.c (hunk 2 above)
Comment (hardik05) [reply fuzz/v3name_fuzzer.c]: made changes.

**Comment by hardik05** [comment]
made requested changes based on review.

**Review by kroeckx — APPROVED** [review]
I would drop the fuzz/corpora/v3name/corpus1 file. We're going to create a separate repository for the fuzz corpora.

I've been running this for some time now, but it's not actually finding the CVE on it's own. It probably needs help by providing a good initial file and I didn't have time yet to create one.

**Comment by hardik05** [comment]
> I've been running this for some time now, but it's not actually finding the CVE on it's own. It probably needs help by providing a good initial file and I didn't have time yet to create one.

Are you running it on version <3.0.8? i immediately get a crash without any corpus, see below logs:
`LD_LIBRARY_PATH=/home/parallels/Research/openssl_vulnerable/build_clang/lib  ./v3name_fuzz
INFO: Running with entropic power schedule (0xFF, 100).
INFO: Seed: 2578623658
INFO: Loaded 1 modules   (5 inline 8-bit counters): 5 [0xaaaae8ff2f68, 0xaaaae8ff2f6d), 
INFO: Loaded 1 PC tables (5 PCs): 5 [0xaaaae8ff2f70,0xaaaae8ff2fc0), 
INFO: -max_len is not provided; libFuzzer will not generate inputs larger than 4096 bytes
INFO: A corpus is not provided, starting from an empty corpus
#2      INITED cov: 3 ft: 4 corp: 1/1b exec/s: 0 rss: 66Mb
#1488   NEW    cov: 5 ft: 6 corp: 2/5b lim: 17 exec/s: 0 rss: 70Mb L: 4/4 MS: 4 ChangeBit-ShuffleBytes-ChangeBit-InsertRepeatedBytes-
#1569   REDUCE cov: 5 ft: 6 corp: 2/4b lim: 17 exec/s: 0 rss: 70Mb L: 3/3 MS: 1 EraseBytes-
#1727   REDUCE cov: 5 ft: 6 corp: 2/3b lim: 17 exec/s: 0 rss: 70Mb L: 2/2 MS: 3 ChangeByte-CrossOver-EraseBytes-
=================================================================
==10847==ERROR: AddressSanitizer: heap-buffer-overflow on address 0xffff9630d3d0 at pc 0xffffa3f33a50 bp 0xffffd8e955a0 sp 0xffffd8e95598
READ of size 4 at 0xffff9630d3d0 thread T0
    #0 0xffffa3f33a4c in ASN1_STRING_cmp /home/parallels/Research/openssl_vulnerable/crypto/asn1/asn1_lib.c:389:13
    #1 0xffffa3f1b544 in ASN1_TYPE_cmp /home/parallels/Research/openssl_vulnerable/crypto/asn1/a_type.c:101:18
    #2 0xffffa4b0f128 in GENERAL_NAME_cmp /home/parallels/Research/openssl_vulnerable/crypto/x509/v3_genn.c:101:18
    #3 0xaaaae8fab900 in LLVMFuzzerTestOneInput /home/parallels/Research/openssl_vulnerable/v3name_fuzz.c:31:5
    #4 0xaaaae8ee5688 in fuzzer::Fuzzer::ExecuteCallback(unsigned char const*, unsigned long) (/home/parallels/Research/openssl_vulnerable/v3name_fuzz+0x45688) (BuildId: 095668975db06a99d77004db3979f1ac063163ed)
    `

**Comment by hardik05** [comment]
will it require CLA so that it can be merged or it can be trivial? let me know.

**Comment by kroeckx** [comment]
It will require a CLA

**Comment by kroeckx** [comment]
> Are you running it on version <3.0.8? i immediately get a crash without any corpus, see below logs:

I've reverted the wrong patch to test it.

**Comment by mattcaswell** [comment]
CLA check is failing because the email address in the commit author field is different to that specified in the CLA. If you amend the author in the commits to match it should start passing.

**Comment by mattcaswell** [comment]
The problem seems to be commit 681f15caba (title "changed name").

While standing in a checkout of your local branch. You can do this:

````
$ git rebase -i HEAD~12
````
In the editor that appears, modify the line for the incorrect commit to change the word "pick" to "e" and then quit the editor.

Then:

````
$ git commit --amend --author "insert-your-email-here"
$ git rebase --continue
````
Then force push the branch to your github remote.

There also seems to be some other spurious commits included in this PR. One related to CMP and one related sparse arrays. Not sure that was intentional?

**Comment by hardik05** [comment]
done changes, good to merge. Please merge it.

**Review by paulidale — APPROVED** [review]

**Review by paulidale — APPROVED** [review]
I did wonder about these going missing....

**Comment by paulidale** [comment]
Merged, thanks for the contribution.

**Pull request closed without merging**